src/m365/pp/commands/managementapp/managementapp-add.ts

```diff
@@ -0,0 +1,130 @@
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+export interface Options extends GlobalOptions {
+  appId?: string;
+  objectId?: string;
+  name?: string;
+}
+
+class PpManagementappAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.MANAGEMENTAPP_ADD;
+  }
+
+  public get description(): string {
+    return 'Register management application for Power Platform';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.appId = typeof args.options.appId !== 'undefined';
+    telemetryProps.objectId = typeof args.options.objectId !== 'undefined';
+    telemetryProps.name = typeof args.options.name !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getAppObjectId(args)
+      .then((appId: string): Promise<void> => {
```

Comment:
If the request in this promise returns response, then we should use `Promise<any>`. `Promise<void>` as a return type indicates that there is no response in which case, we shouldn't have `logger.log(res)` in the next section, because it would print an empty line.